lang/src/main/java/com/sap/psr/vulas/backend/BackendConnector.java

```diff
@@ -588,7 +582,7 @@ public void uploadLibraryFile(String _sha1, Path _file) throws BackendConnection
 	 * @return
 	 * @throws BackendConnectionException
 	 */
-	public boolean uploadGoalExecution(GoalContext _ctx, AbstractGoal _gexe) throws BackendConnectionException {
+	public boolean uploadGoalExecution(GoalContext _ctx, AbstractGoal _gexe, boolean _before) throws BackendConnectionException {
```

Comment:
Why do we need to distinguish the upload of the goal _before the goal execution ? wasn't it ok when it was based on the application existence?

Reply:
When testing different read/write configs, I noticed that the `HttpRequestList` is not properly serialized when the mode is `READ_ONLY` or `OFFLINE`. Only the first of the requests added to the list got serialized. That is why I changed from using the list to using a conditional `PUT` or `POST`, depending on the boolean argument.